File: homework_2/printPrime.c
#include <stdio.h>
#include <stdbool.h>

bool isPrime(int initial) {
        bool prime = true;

        for(int i = 2; i <= initial / 2; i++) {
                if(initial > 2 && initial % i == 0) {
                        prime = false;
                }
        }

        return prime;
}

int main() {
        int number;


        printf("Enter an integer: ");
        scanf("%d", &number);

        for(int i = 2; i <= number; i++) {
                if(isPrime(i)) {
                        printf("%d \n", i); //loop through every number from 2 to the number and call the isPrime function for each
                }
        }

        return 0;
}

File: homework_2/printPrime.py
def isPrime(number):
    result = True
    i = 2
    
    if number == 0 or number == 1:
        result = False
    elif number == 2:
        return result
    else:
        while i < number / 2 + 1:
            if number % i == 0:
                result = False
                
            i += 1
        
    return result

def main():
    number = int(input("Enter a number: "))
    
    for i in range (0, number, 1):
        result = isPrime(i)
        if result == True:
            print(i)
    
    return 0
    
if __name__ == "__main__":
    main()    
